api/v1alpha1/helmchartproxy_types.go

```diff
@@ -65,6 +65,14 @@ type HelmChartProxySpec struct {
 	// include options such as wait, skipCRDs, timeout, waitForJobs, etc.
 	// +optional
 	Options *HelmOptions `json:"options,omitempty"`
+
+	// CredentialsPath is the path to the OCI credentials file. If it is not specified, the default path will be used.
```

Comment:
I think in the case of private Helm repositories (e.g GitHub or even things that are plugin specific like S3) the credentials are very much provider-specific (GitHub you need to have the GitHub token as the environment variable, S3 you need AWS credentials either in env or in the AWS ~/.aws/credentials file etc...).
The reason why this is OCI specific is that it's provider agnostic so it's possible to come up with a generic solution. Though I don't see why the same CRD spec couldn't be reused to fetch secrets that store provider-specific credentials in the case of repositories.

Reply:
Gotcha, I'm not too familiar with how private Helm repo auth works. I think a secret makes the most sense, but what I'm hoping to avoid is having to add a spec field for OCI, another for GitHub, another for AWS, etc. If a secret can support all those cases (since the contents of that secret can be anything), I think that would be ideal.